ovs/extensions/support/agent.py

```diff
@@ -195,16 +199,35 @@ def _set_client(self):
         """
         if self._client is None:
             try:
-                self._client = SSHClient(endpoint='127.0.0.1')
+                self._client = SSHClient(endpoint='127.0.0.1', username='root')
             except Exception:
                 self.logger.exception('Could not instantiate a local client')
         return self._client
 
     def _get_package_information(self):
-        versions_dict = collections.OrderedDict()
-        for pkg_name, version in self._package_manager.get_installed_versions(client=self._client).iteritems():
-            versions_dict[pkg_name] = str(version)
-        return versions_dict
+        versions_dict = {self._client.ip: {}}
+        # ALba is always installed with OpenvStorage. The current split however offloads retrieving Alba information to the AlbaNode which is not
+        # present for non openvstorage-hc installs. Therefore explicititely request the alba information like this (otherwise update will get jeopardized).
+        final_dict = {}
+        threads = []
+
+        for fct in Toolbox.fetch_hooks(component='update', sub_component='get_package_info_cluster'):
+            thread = Thread(target=fct, args=(self._client, versions_dict))
+            thread.start()
+            threads.append(thread)
+
+        for thread in threads:
+            thread.join()
+
+        for versions in versions_dict[self._client.ip].itervalues():
+            for package, version in versions.iteritems():
+                str_version = str(version)
+                if package in final_dict:
+                    if str_version != final_dict[package]:
+                        final_dict[package] = min(version, final_dict[package])
```

Comment:
This is comparing strings. Might not give the result you want.
Instead do the followinging: build your dict as you are now but keep the looseversion item in it.
Before returning: covert all looseversions to strings

Reply:
Its a min(LooseVersion(), str)